### ISSUE TITLE
[MSVC] Build failure after enabling /permissive- compilation option

### ISSUE THREAD

**Issue by NEIL-smtg** [x64dbg/x64dbg#3447]
### Operating System

Windows 10 x64 bit

### x64dbg Version

Latest commit

### Describe the issue

Recently, as I tried to build x64dbg from source with MSVC, I have encountered the errors as following:

```c++
C:\gitP\x64dbg\x64dbg\src\dbg\_dbgfunctions.h(187,9): error C3861: 'getField': identifier not found [C:\gitP\x64dbg\x64dbg\src\dbg\x64dbg_dbg.vcxproj]
(compiling source file '../dbg/analysis/analysis.cpp')
C:\gitP\x64dbg\x64dbg\src\dbg\_dbgfunctions.h(187,17): 'getField': function declaration must be available as none of the arguments depend on a template parameter
C:\gitP\x64dbg\x64dbg\src\dbg\_dbgfunctions.h(184,10): This diagnostic occurred in the compiler generated function 'void <unnamed-tag>::GetField(BP_FIELD,T &)'
C:\gitP\x64dbg\x64dbg\src\dbg\_dbgfunctions.h(170,1): error C5208: unnamed class used in typedef name cannot declare members other than non-static data members, member enumerations, or member classes [C:\gitP\x64dbg\x64dbg\src\dbg\x64dbg_dbg.vcxproj]
(compiling source file '../dbg/analysis/analysis.cpp')
```

### Steps to reproduce

1. git clone https://github.com/x64dbg/x64dbg.git
2. cd x64dbg/x64dbg
3. git submodule sync
4. git submodule update --init --recursive
5. mkdir build_amd64
6. cd build_amd64
7. set \_CL\_=/permissive-
8. msbuild /m /p:Platform=x64 /p:Configuration=Release ../x64dbg.sln /t:Rebuild /p:PlatformToolset=v143 /p:OutDir=C:\gitP\x64dbg\x64dbg\build_amd64 2>&1

### Attachments

Build log: [dbg_log.txt](https://github.com/user-attachments/files/17218998/dbg_log.txt)

**Comment by mrexodia** [comment]
Thanks, but why are you compiling with `/permissive-`?

**Comment by NEIL-smtg** [comment]
@mrexodia Actually I work on MSVC compiler testing, and we regularly build popular open-source project including x64dbg, with development builds of MSVC along with cl options such as /permissive- or /std:c++latest  in order to find and fix regressions before they ship and cause problems for you.

**Comment by mrexodia** [comment]
I fixed the error, let me know if it's still failing! I ran into an issue when upgrading the toolset manually, keeping the SDK to 7.0. Probably this is an unsupported scenario, but it will start failing `#include <assert.h>` in C files...

And FYI, once #3413 gets merged this project will eventually deprecate the visual studio solution. Might want to subscribe to that in case that causes problems for your tests...
